File: frontend/src/services/internalBidDemand.ts
import api from './api'

export interface BidItem {
  pkg: string        // 包号，如 "包1"
  no: string         // 编号，如 "1-1"
  name: string       // 标的名称
  unit_price: string // 单价（元）
  qty: string        // 数量
  unit: string       // 计量单位
  amount: string     // 标的金额（元），= unit_price * qty
}

export interface BidPackage {
  pkg_name: string           // 合同包名称
  budget: string             // 预算金额
  limit_price: string        // 最高限价
  review_method: string      // 评审方法：综合评分法/最低评标价法/性价比法
  price_method: string       // 定价方式：固定单价/固定总价
  allow_joint: string        // 是否支持联合体投标：是/否
  allow_subcontract: string  // 是否允许合同分包：是/否
}

export interface ReviewFactor {
  factor: string      // 评审因素
  score: string       // 评审分值
  is_objective: string // 是否客观项：是/否/空
  standard: string    // 评审标准
}

export interface InternalBidDemand {
  id: number
  project_id: number | null

  // 表头
  demand_dept: string
  manage_dept: string

  // 第一部分
  year: string
  compile_date: string
  category: string           // 货物/服务/工程
  overview: string
  has_consultant: number     // 0=否 1=是
  consultant_note: string

  // 第三部分
  proc_method: string        // 院内竞选/院内单一来源
  pkg_split: string          // 不分包采购/分包采购
  multi_year: number         // 0=否 1=是

  // 第四部分
  items_json: string         // JSON string
  packages_json: string      // JSON string

  // 第五/六部分
  tech_req: string
  biz_req: string

  // 第七部分
  special_qual: string

  // 第八部分
  review_factors_json: string // JSON string

  // 第九部分
  contract_type: string      // 多选逗号分隔
  actual_settlement: number  // 0=否 1=是
  contract_period: string
  contract_location: string
  payment_terms: string
  acceptance_standard: string
  warranty: string
  ip_ownership: string
  cost_risk: string
  breach_dispute: string
  contract_other: string
  perf_bond: number          // 0=否 1=是
  perf_bond_ratio: string
  perf_bond_method: string
  perf_bond_note: string
  quality_bond: number       // 0=否 1=是

  // 第十部分
  accept_org: string         // 自行验收/委托采购代理机构验收
  invite_supplier: number
  invite_expert: number
  invite_service: number
  invite_third: number
  accept_procedure: string
  accept_time: string
  accept_other: string
  tech_accept: string
  biz_accept: string
  accept_std: string
  accept_other2: string

  status: string             // 初稿/定稿
  created_by: string
  created_at: string
  updated_at: string

  // enriched fields
  project_name?: string
  project_number?: string
  budget_amount?: number
}

export const listInternalBidDemands = (params?: { project_id?: number; status?: string }) =>
  api.get<{ ok: boolean; data: InternalBidDemand[] }>('/internal-bid-demands', { params })

export const getInternalBidDemand = (id: number) =>
  api.get<{ ok: boolean; data: InternalBidDemand }>(`/internal-bid-demands/${id}`)

export const createInternalBidDemand = (data: Partial<InternalBidDemand>) =>
  api.post<{ ok: boolean; data: InternalBidDemand }>('/internal-bid-demands', data)

export const updateInternalBidDemand = (id: number, data: Partial<InternalBidDemand>) =>
  api.put<{ ok: boolean; data: InternalBidDemand }>(`/internal-bid-demands/${id}`, data)

export const deleteInternalBidDemand = (id: number) =>
  api.delete<{ ok: boolean }>(`/internal-bid-demands/${id}`)

export const finalizeInternalBidDemand = (id: number) =>
  api.post<{ ok: boolean; message: string }>(`/internal-bid-demands/${id}/finalize`)

export const revokeInternalBidDemand = (id: number) =>
  api.post<{ ok: boolean; message: string }>(`/internal-bid-demands/${id}/revoke`)

export const internalBidDemandWordUrl = (id: number) =>
  `/api/internal-bid-demands/${id}/word`

File: backend/models/internal_bid_demand.py
from models import db


class InternalBidDemand(db.Model):
    __tablename__ = "internal_bid_demands"
    id = db.Column(db.Integer, primary_key=True)
    project_id = db.Column(db.Integer, db.ForeignKey("projects.id"), nullable=True)

    # 表头（院内竞选需求是「项目的起点」，项目名称/预算在此手填，不再关联已立项的项目）
    project_name = db.Column(db.String(200), default="")      # 项目名称（手填）
    budget_amount = db.Column(db.Float, nullable=True)        # 1.6 预算金额（手填，元）
    demand_dept = db.Column(db.String(100), default="")       # 需求科室
    manage_dept = db.Column(db.String(100), default="")       # 归口管理科室

    # 第一部分
    year = db.Column(db.String(10), default="")               # 1.2所属年度
    compile_date = db.Column(db.String(30), default="")       # 1.4编制时间
    category = db.Column(db.String(10), default="货物")       # 1.5 货物/服务/工程
    overview = db.Column(db.Text, default="")                 # 1.7项目概况
    has_consultant = db.Column(db.Integer, default=0)         # 1.8 0=否 1=是
    consultant_note = db.Column(db.Text, default="")          # 1.8 若是填写

    # 第三部分
    proc_method = db.Column(db.String(20), default="院内竞选")  # 3.1 院内竞选/院内单一来源
    pkg_split = db.Column(db.String(20), default="不分包采购")  # 3.2 不分包采购/分包采购
    multi_year = db.Column(db.Integer, default=0)              # 3.3 0=否 1=是

    # 第四部分
    items_json = db.Column(db.Text, default="[]")              # 标的情况 JSON
    packages_json = db.Column(db.Text, default="[]")           # 分包情况 JSON

    # 第五/六部分
    tech_req = db.Column(db.Text, default="")                  # 技术要求
    biz_req = db.Column(db.Text, default="")                   # 商务要求

    # 第七部分
    special_qual = db.Column(db.Text, default="")              # 特殊资格要求

    # 第八部分
    review_factors_json = db.Column(db.Text, default="[]")     # 评审因素 JSON

    # 第九部分
    contract_type = db.Column(db.String(200), default="")      # 合同类型（多选逗号分隔）
    actual_settlement = db.Column(db.Integer, default=0)       # 是否据实结算 0=否 1=是
    contract_period = db.Column(db.String(200), default="")    # 合同履行期限
    contract_location = db.Column(db.String(200), default="")  # 合同履约地点
    payment_terms = db.Column(db.Text, default="")             # 合同支付约定
    acceptance_standard = db.Column(db.Text, default="")       # 验收交付标准和方法
    warranty = db.Column(db.Text, default="")                  # 质量保修范围和保修期
    ip_ownership = db.Column(db.Text, default="")              # 知识产权归属
    cost_risk = db.Column(db.Text, default="")                 # 成本补偿和风险分担约定
    breach_dispute = db.Column(db.Text, default="")            # 违约责任与解决争议的方法
    contract_other = db.Column(db.Text, default="")            # 合同其他条款
    perf_bond = db.Column(db.Integer, default=0)               # 履约保证金 0=否 1=是
    perf_bond_ratio = db.Column(db.String(100), default="")    # 比例
    perf_bond_method = db.Column(db.String(100), default="")   # 缴纳方式
    perf_bond_note = db.Column(db.Text, default="")            # 缴纳说明
    quality_bond = db.Column(db.Integer, default=1)            # 质量保证金 默认是

    # 第十部分
    accept_org = db.Column(db.String(30), default="自行验收")   # 验收组织方式
    invite_supplier = db.Column(db.Integer, default=0)
    invite_expert = db.Column(db.Integer, default=0)
    invite_service = db.Column(db.Integer, default=0)
    invite_third = db.Column(db.Integer, default=0)
    accept_procedure = db.Column(db.Text, default="")
    accept_time = db.Column(db.Text, default="")
    accept_other = db.Column(db.Text, default="")
    tech_accept = db.Column(db.Text, default="")
    biz_accept = db.Column(db.Text, default="")
    accept_std = db.Column(db.Text, default="")
    accept_other2 = db.Column(db.Text, default="")

    status = db.Column(db.String(10), default="初稿")          # 初稿/定稿
    created_by = db.Column(db.String(50), default="")
    created_at = db.Column(db.String(30), default="")
    updated_at = db.Column(db.String(30), default="")

    def to_dict(self):
        return {c.name: getattr(self, c.name) for c in self.__table__.columns}
